pkg/compiler/func_scope.go

```diff
@@ -85,7 +85,19 @@ func (c *codegen) getFuncNameFromDecl(pkgPath string, decl *ast.FuncDecl) string
 		case *ast.Ident:
 			name = t.Name + "." + name
 		case *ast.StarExpr:
-			name = t.X.(*ast.Ident).Name + "." + name
+			switch t.X.(type) {
+			case *ast.Ident:
+				name = t.X.(*ast.Ident).Name + "." + name
+			case *ast.IndexExpr:
+				// Generic func declaration receiver: func (x *Pointer[T]) Load() *T
+				name = t.X.(*ast.IndexExpr).X.(*ast.Ident).Name + "." + name
+			}
```

Comment:
Panic by default?

Reply:
No, we won't panic here. We will retrieve the proper name and I want this code to be kept in future (unlike the rest of commits in this PR). We will analize whether the method is generic in a separate place.

Reply:
Fixed.